test/unit/packwerk/commands/update_todo_command_test.rb

```diff
@@ -77,6 +77,74 @@ class UpdateTodoCommandTest < Minitest::Test
         refute result
       end
 
+      test "#run returns exit code 1 when there are violations of strict mode" do
+        use_template(:minimal)
+
+        source_package = Packwerk::Package.new(
+          name: "components/source",
+          config: { "enforce_dependencies" => "strict" },
+        )
+        write_app_file("#{source_package.name}/package_todo.yml", <<~YML.strip)
+          ---
+          "components/destination":
+            "::SomeName":
+              violations:
+              - dependency
+              files:
+              - components/source/some/path.rb
+        YML
+
+        past_offense = ReferenceOffense.new(
+          reference: build_reference(path: "components/source/some/path.rb", source_package: source_package),
+          message: "some message",
+          violation_type: ReferenceChecking::Checkers::DependencyChecker::VIOLATION_TYPE
+        )
+        new_offense = ReferenceOffense.new(
+          reference: build_reference(path: "components/source/other/path.rb", source_package: source_package),
+          message: "other message",
+          violation_type: ReferenceChecking::Checkers::DependencyChecker::VIOLATION_TYPE
+        )
+
+        RunContext.any_instance.stubs(:process_file).returns([past_offense, new_offense])
+        FilesForProcessing.any_instance.stubs(:files).returns(Set.new(["components/source/some/path.rb"]))
+
+        out = StringIO.new
+        configuration = Configuration.from_path
+        update_command = UpdateTodoCommand.new(
+          [],
+          configuration: configuration,
+          out: out,
+          err_out: StringIO.new,
+          progress_formatter: Formatters::ProgressFormatter.new(out),
+          offenses_formatter: configuration.offenses_formatter
+        )
+
+        result = update_command.run
+
+        expected_output = <<~EOS
+          📦 Packwerk is inspecting 1 file
+
+          📦 Finished in \\d+\\.\\d+ seconds
+
+          components/source/some/path.rb
+          some message
+
+          components/source/other/path.rb
+          other message
+
+          2 offenses detected
```

Comment:
It feels like this message should say something about not being able to successfully add certain violations to the TODO list due to strict mode. Right now it's unclear why it wasn't added.

Reply:
You're right, I've amended the message to only include strict violations that are blocking for the `update_todo` command to complete successfully and changed the final message to explain that something went wrong.